eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -2298,6 +2298,17 @@ default String makeString(String start, String separator, String end)
         return stringBuilder.toString();
     }
 
+    /**
+     * Returns a string representation of the collection, created by applying the function supplied to each element,
+     * with the elements separated by the specified separator and enclosed between the start and end strings.
+     *
+     * @return a string representation of the mapped collection
+     */
+    default String makeString(Function<? super T, String> function, String start, String separator, String end)
+    {
+        return this.collect(function).makeString(start, separator, end);
```

Comment:
This should be `this.asLazy().collect(function).makeString(start, separator, end)`. The call to `collect` here will create a temporary collection otherwise in most cases except for `LazyIterable`.

Reply:
Sorted